pennylane/templates/subroutines/qram.py

```diff
@@ -0,0 +1,359 @@
+        m = next(iter(m_set))
+        bitstrings = list(bitstrings)
+
+        qram_wires = Wires(qram_wires)
+
+        n_k = len(qram_wires)
+        if (1 << n_k) != len(bitstrings):
+            raise ValueError("len(bitstrings) must be 2^(len(qram_wires)).")
+
+        target_wires = Wires(target_wires)
+        if m != len(target_wires):
+            raise ValueError("len(target_wires) must equal bitstring length.")
+
+        bus_wire = Wires(work_wires[0])
+        divider = len(work_wires[1:]) // 3
+        dir_wires = Wires(work_wires[1 : 1 + divider])
+        portL_wires = Wires(work_wires[1 + divider : 1 + divider * 2])
+        portR_wires = Wires(work_wires[1 + divider * 2 : 1 + divider * 3])
+
+        expected_nodes = (1 << n_k) - 1 if n_k > 0 else 0
+
+        if len(work_wires) != 1 + 3 * expected_nodes:
+            raise ValueError(f"work_wires must have length {1 + 3 * expected_nodes}.")
+
+        all_wires = (
+            list(qram_wires)
+            + list(target_wires)
+            + list(bus_wire)
+            + list(dir_wires)
+            + list(portL_wires)
+            + list(portR_wires)
+        )
+
+        wire_manager = _QRAMWires(qram_wires, target_wires, bus_wire, dir_wires, portL_wires, portR_wires)
+
+        self._hyperparameters = {
+            "wire_manager": wire_manager,
+            "m": m,
+            "n_k": n_k,
+            "bitstrings": bitstrings,
+        }
+
+        super().__init__(wires=all_wires, id=id)
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    def _path_ctrls(self, i_low: int):
+        """(controls, values) for the router path to leaf `i_low` (MSB-first across n_k)."""
+        ctrls, vals = [], []
+        wire_manager = self.hyperparameters["wire_manager"]
+        n_k = self.hyperparameters["n_k"]
+        for k in range(n_k):
+            prefix = i_low >> (n_k - k)
+            ctrls.append(wire_manager.router(k, prefix))
+            vals.append((i_low >> (n_k - 1 - k)) & 1)
+        return ctrls, vals
+
+    # ---------- Address Loading via CSWAP routing ----------
+    def _mark_routers_via_bus(self) -> list:
+        """Write low-order address bits into router directions **layer-by-layer** via the bus.
+
+        For each low bit a_k (k = 0..n_k-1):
+          1) SWAP(qram_wires[k], bus)
+          2) Route bus down k levels (CSWAPs controlled by routers at levels < k)
+          3) At node (k, path-prefix), SWAP(bus, dir[k, path-prefix])
+        """
+        ops = []
+        wire_manager = self.hyperparameters["wire_manager"]
+        for k in range(self.hyperparameters["n_k"]):
+            # 1) load a_k into the bus
+            ops.append(
+                SWAP(
+                    wires=[
+                        wire_manager.qram_wires[k],
+                        wire_manager.bus_wire[0],
+                    ]
+                )
+            )
```

Comment:
```suggestion
            origin = wire_manager.qram_wires[k]
            target = wire_manager.bus_wire[0]
            ops.append(SWAP((origin, target)))
```
same for the various other similar uses.  Makes things a bit more readable.